app/models/goods_nomenclature.rb

```diff
@@ -225,12 +225,28 @@ def chapter?
     !!goods_nomenclature_item_id.match(/\A\d{2}00000000\z/)
   end
 
+  def non_grouping?
+    producline_suffix == GoodsNomenclatureIndent::NON_GROUPING_PRODUCTLINE_SUFFIX
+  end
+
+  def grouping?
+    !non_grouping?
+  end
+
   def path_declarable?
-    path_children.none? && producline_suffix == GoodsNomenclatureIndent::NON_GROUPING_PRODUCTLINE_SUFFIX
+    non_grouping? && path_children_dataset.limit(1).none?
```

Comment:
I'm aware that the path_children association isn't eager loadable at present. I'm sure we could make these associations eager loadable but it feels like this is waisted effort right now since its getting removed/replaced by nested set which works with the Time Machine